src/pyscipopt/scip.pyx

```diff
@@ -1532,6 +1533,79 @@ cdef class Model:
         PY_SCIP_CALL(SCIPcreateSol(self._scip, &solution.sol, _heur))
         return solution
 
+    def printBestSol(self, write_zeros=False):
+        """Prints the best feasible primal solution."""
+        PY_SCIP_CALL(SCIPprintBestSol(self._scip, NULL, write_zeros));
+
+    def printSol(self, Solution solution, write_zeros=False):
+      """Print the given primal solution.
+
+      Keyword arguments:
+      solution -- solution to print
+      write_zeros -- include variables that are set to zero
+      """
+      PY_SCIP_CALL(SCIPprintSol(self._scip, solution.sol, NULL, write_zeros));
+
+    def writeBestSol(self, filename="origprob.sol", write_zeros=False):
+        """Write the best feasible primal solution to a file.
+
+        Keyword arguments:
+        filename -- name of the output file
+        write_zeros -- include variables that are set to zero
+        """
+        # use this doubled opening pattern to ensure that IOErrors are
+        #   triggered early and in Python not in C,Cython or SCIP.
+        with open(filename, "w") as f:
+            cfile = fdopen(f.fileno(), "w")
+            PY_SCIP_CALL(SCIPprintBestSol(self._scip, cfile, write_zeros));
```

Comment:
This way of implementing file operations looks nice. What do you think, @fserra?

Reply:
I like it. I guess it is alright if IO errors are caught by python rather than SCIP.